### PR TITLE
Support custom network in rosetta bdd client

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds network nodes customization support to the rosetta bdd test client

-  Add a server `network` configuration to customize nodes for the main network
- Simplify configuration parsing for rosetta server
- Update bdd test client README

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rosetta/test/bdd-client/client/client.go [hunk 1]
```diff
@@ -353,11 +353,16 @@ func NewClient(serverCfg Server, operators []Operator) Client {
 	}
 
 	// create hedera client
-	hederaClient, err := hedera.ClientForName(network.Network)
-	if err != nil {
-		log.Fatalf("Failed to create client for hedera '%s'", network.Network)
+	var hederaClient *hedera.Client
+	if len(serverCfg.Network) == 0 {
+		if hederaClient, err = hedera.ClientForName(network.Network); err != nil {
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/test/bdd-client/client/client.go:358]: The same config property supports both string and map?

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-rosetta/test/bdd-client/client/client.go (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rosetta/test/bdd-client/client/client.go:358]: nope. `network.Network` is the value read from the rosetta API `/network/list`, `serverCfg` is from configuration file. I think that's the confusion here?

Comment (steven-sheehy) [reply hedera-mirror-rosetta/test/bdd-client/client/client.go:358]: Got it

**Review by ijungmann — APPROVED** [review]

**Pull request merged**